components/sup/src/manager/mod.rs

```diff
@@ -650,12 +741,7 @@ impl Manager {
     fn start_initial_services_from_watcher(&mut self) -> Result<()> {
         for service_event in self.watcher.initial_events()? {
             match service_event {
-                SpecWatcherEvent::AddService(spec) => {
```

Comment:
I don't think we should load services which are supposed to be in the `Down` state, I'd keep this block as is. The code makes assumptions that a "loaded" service has an active rumor which is shared along with participating in the tick loop. I think this complicates that logic by needing to short circuit things with a state check.

It's worth noting that right now Adam is working on some logic which will allow us to tombstone service rumors so when you stop a service it actually let's peers know that it went away